core/src/main/scala/org/apache/spark/storage/DiskBlockObjectWriter.scala

```diff
@@ -81,12 +82,19 @@ private[spark] class DiskBlockObjectWriter(
    */
   private var numRecordsWritten = 0
 
+  private var sparkConf: SparkConf = null
```

Comment:
Why not make this a constructor argument? You can have an overloaded constructor so you don't have to change all the existing call sites. Then this can be a `val`, and you don't need the setter.